lib/src/push/push_manager.dart

```diff
@@ -68,9 +67,8 @@ class PushManager {
 
   PushManager._internal();
 
-  void setup(BuildContext buildContext) async {
-    this._buildContext = buildContext;
-    _pushBloc = BlocProvider.of<PushBloc>(_buildContext);
+  Future<void> setup(PushBloc pushBloc) async {
```

Comment:
```suggestion
  Future<void> setupAsync(PushBloc pushBloc) async {
```
it is an async function so the naming should be "...Async"